conans/client/cmd/export_pkg.py

```diff
@@ -43,6 +43,11 @@ def export_pkg(app, recorder, full_ref, source_folder, build_folder, package_fol
 
     recipe_hash = layout.recipe_manifest().summary_hash
     conanfile.info.recipe_hash = recipe_hash
+
+    for package_name, env_vars in conanfile._conan_env_values.data.items():
```

Comment:
Looks good, please improve with  small refactors:
- Code is almost duplicated from ``def _propagate_info(node):``. Extract the logic to a common place, and call it from here and ``_propagate_info()``. I think that the filtering of information from dependencies might be necessary too.
- Call the new method closer to the above ``load_deps_info()`` call. Wrap it in a new private function here in this file that could be called something like ``_init_conanfile_infos()``